eventstream.go

```diff
@@ -291,8 +291,14 @@ func (p *PublishBuilder) Payload(payload map[string]interface{}) *PublishBuilder
 	return p
 }
 
-// ErrorCallback function to handle the event when failed to publish
-func (p *PublishBuilder) ErrorCallback(errorCallback func(event *Event, err error)) *PublishBuilder {
+// ErrorCallback function to handle the event when failed to publish.
+//
+// Deprecated: As the underlying client handles error asynchronously, it's not possible to pass the Event to the
```

Comment:
If this is deprecated, is there a new recommended way to handle errors? 
I think we should just remove this field if there's no real use for it. If the callback gets a nil event where it previously didn't expect it, it will crash. Better just remove it from the API then

Reply:
If error handling is needed, we can use PublishSync instead, with cost of slower produce rate. Yup, I'm thinking about removing it as well, wdyt @thoriqsatriya ?

Reply:
> If error handling is needed, we can use PublishSync instead
Yep agree with this. Let's remove this